ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceAlgorithms.java

```diff
@@ -61,19 +63,21 @@ private static <TInner, TWrapper extends ResourceWithSas<TInner>, TOut> TOut res
                     }
                 }, actionName, attributes);
             } catch (Exception e) {
+                ex = e;
                 log.warn(String.format("Error during retry %d of %d for %s", i + 1, RETRY_COUNT, actionName), e);
             }
         }
-        throw new IngestionClientException(String.format("%s: All %d retries failed - used resources: %s", actionName, RETRY_COUNT,
-                totalAttributes.stream().map(x -> String.format("%s (%s)", x.get("resource"), x.get("account"))).collect(Collectors.joining(", "))));
+        throw new IngestionClientException(String.format("%s: All %d retries failed with last error: %s\n. Used resources: %s", actionName, RETRY_COUNT,
+                totalAttributes.stream().map(x -> String.format("%s (%s)", x.get("resource"), x.get("account"))).collect(Collectors.joining(", ")),
+                ExceptionsUtils.getMessageEx(ex)));
     }
 
     public static void postToQueueWithRetries(ResourceManager resourceManager, AzureStorageClient azureStorageClient, IngestionBlobInfo blob)
-            throws IngestionClientException, IngestionServiceException {
+            throws IngestionClientException, IngestionServiceException, JsonProcessingException {
```

Comment:
That's breaking right?

Reply:
no

Reply:
Why not? it's a public method on a public class

Reply:
you are right - im not sure why its public tho